### PR TITLE
Output sccache stats as a notice and a summary table

### PR DESCRIPTION
**Pull request by orf**
Checking the effectiveness of sccache when using this action is a bit annoying: the post action is minimized, and you need to go into each individual job.

This PR adds high-level cache hits/misses/errors as a notice (available at the top of the job summary), and a more detailed table as a job summary.

Screenshot 2024-05-12 at 12 59 18

Screenshot 2024-05-12 at 13 00 16

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -21,11 +21,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
```
Comment (sylvestre) [inline .github/workflows/CI.yml:24]: please revert this changes and do them in a different PR

Comment (orf) [reply .github/workflows/CI.yml:24]: Done! https://github.com/Mozilla-Actions/sccache-action/pull/114

**Comment by sylvestre** [comment]
Great idea btw

**Comment by orf** [comment]
Thanks! I’ve integrated the branch with the project I was working on, you can see some builds on it: https://github.com/orf/pyvector

**Comment by sylvestre** [comment]
could you please rebase it? thanks

**Comment by sylvestre** [comment]
also, where can I see it on pyvector ?

**Comment by orf** [comment]
Rebased! You can see it in the actions tab. For example: https://github.com/orf/pyvector/actions/runs/9051168104

**Comment by sylvestre** [comment]
nice :)
Maybe add % too ?
(should not be too hard math ;)

**Comment by orf** [comment]
Good idea, done! I’ve triggered a build here to see what it looks like: https://github.com/orf/pyvector/actions/runs/9053826212

I don’t have much more time to work on this though, so lemme know if there is anything else you think I should change or add

**Comment by sylvestre** [comment]
Just one request: Could you please add a the % again ? (in the right column)
thanks
image

**Comment by orf** [comment]
Done!

**Comment by sylvestre** [comment]
sorry, i forgot to merge it :)
